src/test/basics/PerfLog_test.cpp

```diff
@@ -288,23 +289,45 @@ class PerfLog_test : public beast::unit_test::suite
                 perms::remove_perms | perms::owner_write |
                 perms::others_write | perms::group_write);
 
-            // Now construct a PerfLog.  The PerfLog should attempt to shut
-            // down the server because it can't open its file.
-            BEAST_EXPECT(parent.stopSignaled == false);
-            auto perfLog {getPerfLog (parent, WithFile::yes)};
-            BEAST_EXPECT(parent.stopSignaled == true);
+            // If the test is running as root, then the write protect may have
+            // no effect.  Make sure write protect worked before proceeding.
```

Comment:
Why not directoy check if we're running as root? Something like:```
            bool const runningAsRoot = [] {
#ifndef _WIN32
                return geteuid() == 0;
#else
                return false;
#endif

Reply:
I actually prefer checking whether the file is writable.

1. It's what matters.  If Perflog can write to the file then the failure won't happen.  So we're looking at what will actually break the test (or not).  The fact that this particular failure was caused by running the test as root is incidental.

2. Beyond that, the check is portable.  I don't have to worry about which operating system I'm running on.

Just my take on it.  But it's not a make-or-break thing for me.

Reply:
I'm fine with that. The downside is we're cluttering up the code that can be one call with something that's more complicated, and we're really dealing with an edge case (running the unit tests as root). But again, I'm fine with keeping it as-is if you prefer it.

Reply:
One more note on this. It can be re-written as:
```
bool const writeProtected =
    !std::ofstream{fullPath.c_str(), std::ios::out | std::ios::app};
```
We don't need `close` as it's called by the destructor, and once that is gone there's no reason to keep the lambda.

Reply:
Write protected is checked twice in this block (lines 283 through 287). How about creating one lambda to check them both. Here's a patch I have in mind: https://github.com/seelabs/rippled/commit/eb72db6b772ebe464db72995a15b1563973670f2

That's clear (`if (writeProtected(fullPath))` is very clear what's happening, and it's "8 additions and 18 deletions", which always make me feel good :-)